cli/utilities/registry.py

```diff
@@ -0,0 +1,180 @@
+    """"""
+    node.exec_command(
+        sudo=True,
+        cmd=f"cp {OPT_CERTS}/domain.crt \
+            /etc/pki/ca-trust/source/anchors/",
+    )
+    node.exec_command(sudo=True, cmd="update-ca-trust")
+    update_cert(node)
+
+
+def copy_cert_to_nodes(node, ceph_nodes):
+    """"""
+    file_name = f"{OPT_CERTS}/domain.crt"
+    for ceph in ceph_nodes:
+        ceph.exec_command(
+            sudo=True, cmd="touch /etc/pki/ca-trust/source/anchors/domain.crt"
+        )
+        contents, _ = node.exec_command(sudo=True, cmd=f"cat {file_name}")
+        key_file = ceph.remote_file(
+            sudo=True,
+            file_name="/etc/pki/ca-trust/source/anchors/domain.crt",
+            file_mode="a",
+        )
+        key_file.write(contents)
+        key_file.flush()
+        ceph.exec_command(sudo=True, cmd="update-ca-trust")
+        update_cert(node)
+
+
+def start_private_registry(node, reg_name):
+    """Create and start the pricate registry on port 5000
+    Args:
+        node(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+    node.exec_command(
+        cmd=f"sudo podman run --restart=always --name {reg_name} \
+                    -p 5000:5000 -v {OPT_DATA}:/var/lib/registry:z \
+                    -v {OPT_AUTH}:/auth:z \
+                    -v {OPT_CERTS}:/certs:z \
+                    -e 'REGISTRY_AUTH=htpasswd' \
+                    -e 'REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm' \
+                    -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+                    -e 'REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt' \
+                    -e 'REGISTRY_HTTP_TLS_KEY=/certs/domain.key' \
+                    -e REGISTRY_COMPATIBILITY_SCHEMA1_ENABLED=true \
+                    -d docker.io/meadityaramteke/library"
+    )
+    log.info("Private Registry Started")
+
+
+def create_images_dict(config):
+    """Return a dictionary with key as image name and value as image path
+    Args:
+        Config(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    image = []
+    supported_overrides = [
+        "node_exporter",
+        "grafana",
+        "prometheus",
+        "alertmanager",
+    ]
+
+    # Fetch images from yaml file
+    node_exporter_image = config.get("ansi_config").get("node_exporter_container_image")
+    grafana_image = config.get("ansi_config").get("grafana_container_image")
+    prometheus_image = config.get("ansi_config").get("prometheus_container_image")
+    alertmanager_image = config.get("ansi_config").get("alertmanager_container_image")
+
+    # Add images to a list
+    image.append(node_exporter_image)
+    image.append(grafana_image)
+    image.append(prometheus_image)
+    image.append(alertmanager_image)
+
+    # Create dictionary of images
+    image_dict = {
+        supported_overrides[i]: image[i] for i in range(len(supported_overrides))
+    }
+    return (image_dict, image)
+
+
+def private_registry_add_image(node, config, reg_user, reg_pass):
```

Comment:
change this to `add_image` and variable names

Reply:
We other functions as well with:
start_private_registry
get_private_images

I would like to keep the name as private_registry_add_image.